scripts/common/preflights.sh

```diff
@@ -657,3 +657,19 @@ function bail_if_unsupported_openebs_to_rook_version() {
     fi
 }
 
+# bail_when_no_object_store_and_s3_enabled will bail if Minio and Rook are not present and kotsadm.s3Disabled is false.
+function bail_when_no_object_store_and_s3_enabled() {
+    if [[ -z "$MINIO_VERSION" && -z "$ROOK_VERSION" ]] && [ -n "$OPENEBS_VERSION" ]; then
+        if [ -n "$KOTSADM_VERSION" ] && [ "$KOTSADM_DISABLE_S3" != "1" ]; then
+             logFail "KOTS with s3 enabled requires an object store."
+             bail "Please, ensure that your installer also provides an object store with either the MinIO or Rook add-on."
+        fi
+        if [ -n "$VELERO_VERSION" ] && [ "$HA_CLUSTER" = "1" ] ; then
```

Comment:
this is not true. it should be the following i believe based on https://github.com/replicatedhq/kURL/blob/61580229c2758349b8ac3efb39e22bdff19298dc/addons/velero/template/base/install.tmpl.sh#L51-L53

```suggestion
        if [ -n "$VELERO_VERSION" ] && [ "$KOTSADM_DISABLE_S3" != "1" ]; then
```